slimmer/slimmer.cpp

```diff
@@ -191,54 +199,62 @@ int parsed_main(int argc, char** argv) {
-                    return lep.global and lep.normChi2 < 10.0 and
-                      lep.nValidMuon > 0 and lep.nMatched > 1 and
-                      lep.nValidPixel > 0 and lep.trkLayersWithMmt > 5 and
-                      lep.dxy < 0.2 and lep.dz < 0.5;
-                  },
-                  pt);
+                    return lep.tight and reliso < 0.15;
+                    // return lep.global and lep.normChi2 < 10.0 and
+                    //   lep.nValidMuon > 0 and lep.nMatched > 1 and
+                    //   lep.nValidPixel > 0 and lep.trkLayersWithMmt > 5 and
+                    //   lep.dxy < 0.2 and lep.dz < 0.5;
+                  });
       }
 
       // Loop over electrons
       for (auto& lep : event.electrons) {
 
         auto abseta = std::abs(lep.eta());
-        auto reliso = lep.combIso()/lep.pt();
+        auto pt = lep.smearedPt;
+        auto reliso = lep.combIso()/pt;
+
+        if(debug::debug)
+          std::cout << "Electron with pt " << lep.pt() << " Corrected to " << pt << " has reliso " << reliso << std::endl;
 
-        check_lep(stored_eles, lep, reliso,
+        check_lep(stored_eles, lep, reliso, pt,
                   [&] {   // Electron preselection
-                    return lep.pt() > 7.0 and abseta < 2.5 and
+                    return lep.smearedPt > 7.0 and abseta < 2.4 and
                       lep.dxy < 0.05 and lep.dz < 0.20 and
                       reliso < 0.4;
                   },
                   [&] {   // Loose electrons for conservative event classification
-                    return lep.mvaWP90;
+                    return true;
+                    // return lep.mvaWP90;
                   },
                   [&] {   // "Medium" electrons, which I define as the cut applied to match MVA training
-                    return lep.pt() > 15.0 and lep.hOverE < 0.09 and reliso < 0.18 and
-                      ((abseta < 1.4442 and lep.sieie < 0.012 and
-                        lep.ecalIso/lep.pt() < 0.4 and lep.hcalIso/lep.pt() < 0.25 and
-                        std::abs(lep.dEtaInSeed) < 0.0095 and std::abs(lep.dPhiIn) < 0.065) or
-                       (abseta > 1.5660 and lep.sieie < 0.033 and
-                        lep.ecalIso/lep.pt() < 0.45 and lep.hcalIso/lep.pt() < 0.28)
-                       );
+                    return true;
+                    // return lep.smearedPt > 15.0 and lep.hOverE < 0.09 and reliso < 0.18 and
+                    //   ((abseta < 1.4442 and lep.sieie < 0.012 and
+                    //     lep.ecalIso/lep.smearedPt < 0.4 and lep.hcalIso/lep.smearedPt < 0.25 and
+                    //     std::abs(lep.dEtaInSeed) < 0.0095 and std::abs(lep.dPhiIn) < 0.065) or
+                    //    (abseta > 1.5660 and lep.sieie < 0.033 and
+                    //     lep.ecalIso/lep.smearedPt < 0.45 and lep.hcalIso/lep.smearedPt < 0.28)
+                    //    );
```

Comment:
@DylanHsu I think we should also include at least these eta cuts for tight lepton counting, though we also want to veto on stuff that goes in there. I'm not sure what is best.

Reply:
from what it looks like in the analysis note, the electron preselection includes the ECAL crack, but the tight object definition does not include the ECAL crack

Reply:
we'll add the cuts you commented out to the MVAWP80 and MVAWP90 bit definition, i.e. here:
https://github.com/sidnarayanan/PandaAnalysis/blob/master/Flat/src/ModulesLepPho.cc#L34-L35